generator/uscore/templates/access_verify_sequence.rb.erb

```diff
@@ -60,6 +65,9 @@ module Inferno
         Organization, Practitioner, PractionerRole, and RelatedPerson.  It also does not test Provenance, as this
         resource type is accessed by queries through other resource types. These resources types are accessed in the more
         comprehensive Single Patient Query tests.
+
+        However, the authorization system must indicate that access is granted to the Encounter, Practitioner and Location
```

Comment:
What's the deal with this?

Reply:
First off, I have a typo in here.  It should be Encounter, Practitioner and Organization (not Location).

This is complicated.  I would love more scrutiny / double checking on this.  As we know, Encounter, Organization, Practitioner, PractitionerRole, Location and RelatedPerson are not directly mapped into USCDI.  However, they are needed in order to completely implement those US Core resources that are directly mapped into USCDI.

Encounter, Organization and Practitioner have at least one 'US Core typed reference' in another US Core profile.  Because of that, you need to support their US Core profile.  And because of that, you really need to support the whole resource, and it makes sense for it to 'stand alone with its own identity', and therefore you need to support the read interaction on it (vs just containing it in resources that reference it).  Therefore, you need to be able to provide scopes for those, either through patient/*.read or explicitly on those resources.  For the scenario where we expect full access, we need to see those scopes.

For PractitionerRole, Location and RelatedPerson, since there are no explicit US core typed references to them in other US Core resources, you can just 'contain' them in their referring resource (similar to Medication).  And therefore we do not need to see their scopes because they may or may not decide to implement them in a standalone way, depending on if those concepts stand alone in their system or not.

Now, to be completely transparent on what this test does, we do check the scopes returned for Practitioner, Organization, and Encounter, but we do not actually perform read requests on ANY of those 6 in this section because it is difficult to find all the needed references.  So we just say 'good enough' on the directly mapped USCDI resources for the actual FHIR searches/reads here, because we do a more thorough job on the complete US Core set of tests.

And FINALLY, what even complicates this further, is that on the 'limited scope' version of all this (vs. the version where we expect to see ALL scopes), the SUT has flexibility to not given the user direct control of authorizing these non-USCDI-mapped resources, so we cannot fail them if they don't provide the Practitioner, Organization and Encounter scopes because it is up to the EHR to decide if given the limited choices the user selected that those need to be included as scopes.

Reply:
I think this makes sense. There are references to these specific profiles, so you have to implement the required RESTful interactions for those profiles. I don't think that the logic behind this is clear in the test details, though.

Reply:
So, just to double check, for Encounter, Organization, Practitioner, PractitionerRole, Location, and RelatedPerson, the expected behavior is to pass the SUT if they provide scopes for these even if they aren't listed in the expected resource grants.

Reply:
For the 'unrestricted access' sequence, we need to see Encounter, Organization, Practitioner in the scopes that are granted during the token exchange.  We've set the rule in that scenario that we want them to show us "everything", and expect to be granted access to everything that has at least read access.

For the 'restricted access sequence', we do not need to see any of those 6 in the scopes that were granted (received).  But we also don't fail them if we DO see them.  Essentially, the SUT can decide when to grant access to them or not based on which subset of USCDI resources were selected by the user.  The user doesn't have to be given the option to explicitly grant 'Organization', for example (though they could).

This is difficult to communicate succinctly.